tests/test_secondary.py

```diff
@@ -0,0 +1,721 @@
+
+TEST_DATA_DIR = os.path.join(uptane.WORKING_DIR, 'tests', 'test_data')
+TEST_DIRECTOR_METADATA_DIR = os.path.join(TEST_DATA_DIR, 'director_metadata')
+TEST_IMAGE_REPO_METADATA_DIR = os.path.join(
+    TEST_DATA_DIR, 'image_repo_metadata')
+TEST_DIRECTOR_ROOT_FNAME = os.path.join(
+    TEST_DIRECTOR_METADATA_DIR, 'root.' + tuf.conf.METADATA_FORMAT)
+TEST_IMAGE_REPO_ROOT_FNAME = os.path.join(
+    TEST_IMAGE_REPO_METADATA_DIR, 'root.' + tuf.conf.METADATA_FORMAT)
+TEST_PINNING_FNAME = os.path.join(TEST_DATA_DIR, 'pinned.json')
+TEMP_CLIENT_DIR_1 = os.path.join(TEST_DATA_DIR, 'temp_test_secondary1')
+TEMP_CLIENT_DIR_2 = os.path.join(TEST_DATA_DIR, 'temp_test_secondary2')
+TEMP_CLIENT_DIR_3 = os.path.join(TEST_DATA_DIR, 'temp_test_secondary3')
+
+# I'll initialize these in the __init__ test, and use this for the simple
+# non-damaging tests so as to avoid creating objects all over again.
+secondary_instance_1 = None # for ECU 'TCUdemocar' in vehicle 'democar'
+secondary_instance_2 = None # for an unknown ECU in vehicle '000'
+secondary_instance_3 = None # for an unknown ECU in vehicle '000'
+
+# Changing these values would require producing new signed test data from the
+# Timeserver (in the case of nonce) or a Secondary (in the case of the others).
+nonce = 5
+vin1 = 'democar'
+vin2 = 'democar'
+vin3 = '000'
+ecu_serial1 = 'TCUdemocar'
+ecu_serial2 = '00000'
+ecu_serial3 = '00000'
+
+# Initialize these in setUpModule below.
+secondary_ecu_key = None
+key_timeserver_pub = None
+key_timeserver_pri = None # simplifies use of test data (re-sign)
+key_directortargets_pub = None
+clock = None
+
+# Set starting firmware fileinfo (that this ECU had coming from the factory)
+factory_firmware_fileinfo = {
+    'filepath': '/secondary_firmware.txt',
+    'fileinfo': {
+        'hashes': {
+            'sha512': '706c283972c5ae69864b199e1cdd9b4b8babc14f5a454d0fd4d3b35396a04ca0b40af731671b74020a738b5108a78deb032332c36d6ae9f31fae2f8a70f7e1ce',
+            'sha256': '6b9f987226610bfed08b824c93bf8b2f59521fce9a2adef80c495f363c1c9c44'},
+        'length': 37}}
+
+expected_updated_fileinfo = {
+    'filepath': '/TCU1.1.txt',
+    'fileinfo': {
+        'custom': {'ecu_serial': 'TCUdemocar'},
+        'hashes': {
+            'sha512': '94d7419b8606103f363aa17feb875575a978df8e88038ea284ff88d90e534eaa7218040384b19992cc7866f5eca803e1654c9ccdf3b250d6198b3c4731216db4',
+            'sha256': '56d7cd56a85e34e40d005e1f79c0e95d6937d5528ac0b301dbe68d57e03a5c21'},
+        'length': 17}}
+
+
+def destroy_temp_dir():
+  # Clean up anything that may currently exist in the temp test directories.
+  for client_dir in [TEMP_CLIENT_DIR_1, TEMP_CLIENT_DIR_2, TEMP_CLIENT_DIR_3]:
+    if os.path.exists(client_dir):
+      shutil.rmtree(client_dir)
+
+
+
+
+
+def setUpModule():
+  """
+  This is run once for the full module, before all tests.
+  It prepares some globals, including two for Secondary ECU client instances.
+  When finished, it will also start up an OEM Repository Server,
```

Comment:
Ah, that's text to be stripped and I've removed it in a commit on another branch to be PR'd soon.